test/e2e/main_test.go

```diff
@@ -120,6 +129,30 @@ func TestMain(m *testing.M) {
 			}
 		}
 
+		props := provisioner.GetProperties(ctx, cfg)
+		var kbsparams string
+		if shouldDeployKbs {
+			log.Info("Deploying kbs")
+			if props["KBS_IMAGE"] == "" || props["KBS_IMAGE_TAG"] == "" {
```

Comment:
This check should happen before not after cluster is set up. Because this can be frustrating. Because if user provides something else or misspell it then the set up fails as follows:

```bash
$ ./caa-provisioner-cli -action=provision
INFO[0000] Creating VPC...
INFO[0000] Creating Resource group test-kartik-pr1518.
INFO[0001] Successfully Created Resource group test-kartik-pr1518.
INFO[0001] Creating Cluster...
INFO[0131] Successfully created federated identity credential "e2e_test_clusterFederatedIdentityCredential" in resource group "test-kartik-pr1518"
INFO[0133] Successfully created federated identity credential "e2e_test_clusterFederatedIdentityCredential" in resource group "test-kartik-pr1518"
INFO[0134] Sync cluster kubeconfig with current config context
INFO[0134] Deploying kbs
FATA[0134] kbs image not provided
```

Reply:
KBS is deployed, but is failing on some config. I am looking into this.

Reply:
```
INFO  kbs] Using config file /etc/kbs/kbs-config.toml
Error: invalid config: missing field `rvps_config`
```

I was getting this error from kbs pod. Are you seeing the same issue ?

Reply:
Yes I had same problem.

Reply:
This check should happen before starting any provisioning.